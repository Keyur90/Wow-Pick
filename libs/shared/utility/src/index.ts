export * from './barcode/GS1BarcodeParser';
export * from './barcode/parseScannedBarcode';
export * from './lib/bags';
export * from './lib/calculateShelfLifeExpiryDate';
export * from './lib/calculateTotalSuppliedQuantity';
export * from './lib/calculateTotalSuppliedQuantityForCollectible';
export * from './lib/calculateTrafficLightStatus';
export * from './lib/camelCase';
export * from './lib/canDispatchOrder';
export * from './lib/constants';
export * from './lib/dates';
export * from './lib/findIndexOfLastUnSuppliedItem';
export * from './lib/getEmptyTotes';
export * from './lib/hasAlreadySuppliedQuantity';
export * from './lib/hooks/useInputValidation';
export * from './lib/hooks/useSecondaryToleranceValidation';
export * from './lib/hooks/useTrafficLightStatus';
export * from './lib/isScannedBarcodeMatchWithArticleBarcode';
export * from './lib/isWeightedQuantityArticle';
export * from './lib/printLabels';
export * from './lib/quantityRange';
export * from './lib/returnToLegacyTrolleyMain';
export * from './lib/units';
export * from './lib/updateSuppliedQuantities';
export * from './lib/validateBarcode';
export * from './lib/validateExpiryDate';
export * from './lib/validateWeightTolerance';
export * from './types/barcode-data';
export * from './types/barcode-type';
export * from './types/BarcodeValidationResponse';
export * from './types/expiry-date-status';
export * from './types/trip-type';
