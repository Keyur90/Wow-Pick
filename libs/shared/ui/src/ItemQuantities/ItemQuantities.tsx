import React from 'react';
import { makeStyles, Theme } from '@material-ui/core/styles';
import { BagStatus } from '@rf2/ui';
import { Tote, ToteItem, Order, Collectible } from '@rf2/picking/data/api-contracts';
import { LabelledValue } from '@rf2/ui';
import { ArticleType } from '@rf2/shared/utility';

interface ItemQuantitiesPropTypes {
  toteItem: ToteItem | Collectible;
  tote: Tote;
  order: Order;
  showExpiry?: boolean;
}

interface StyledProps {
  showExpiry: boolean;
}

const useStyles = makeStyles<Theme, StyledProps>((theme) => ({
  root: {
    background: theme.palette.common.white,
    borderBottom: ({ showExpiry }) => `1px solid ${showExpiry ? '#F3F3F7' : '#E0E0E0'}`,
    display: 'flex',
    flexDirection: 'row',
    '& > div': {
      width: '25%',
      padding: theme.spacing(1),
      borderLeft: '1px solid #F3F3F7',
      '&:first-child': {
        padding: theme.spacing(1, 1, 1, 2),
        borderLeft: 'none',
      },
      '&:last-child': {
        backgroundColor: '#EEEEEE',
      },
    },
  },
  toteStatus: {
    display: 'flex',
    alignItems: 'center',
    '& > span': {
      fontSize: '0.75rem',
      lineHeight: '0.75rem',
      marginLeft: theme.spacing(0.5),
    },
  },
}));

const ItemQuantities: React.FC<ItemQuantitiesPropTypes> = ({ toteItem, tote, order, showExpiry = false }) => {
  const classes = useStyles({ showExpiry });
  const {
    orderedQuantity,
    stockOnHand,
    article: { pricingUnit, useWeightRange },
    totalSuppliedQuantity,
  } = toteItem;
  const { position } = tote;
  const { packagingType } = order;
  const unitOrderDisplay = pricingUnit === ArticleType.EACH || useWeightRange ? `(EA)` : '(KG)';

  return (
    <div className={classes.root}>
      <LabelledValue label={`ORD ${unitOrderDisplay}`} value={orderedQuantity} />
      <LabelledValue
        label="SUP"
        value={totalSuppliedQuantity <= orderedQuantity ? totalSuppliedQuantity : orderedQuantity}
      />
      <LabelledValue label="SOH" value={stockOnHand} />
      <LabelledValue
        label="TOTE"
        value={
          <div className={classes.toteStatus}>
            {position}
            <BagStatus bagType={packagingType} />
          </div>
        }
        emphasis
      />
    </div>
  );
};

export { ItemQuantities };
