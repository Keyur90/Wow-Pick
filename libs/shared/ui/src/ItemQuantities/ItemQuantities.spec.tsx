import { PackagingTypes, PickingZoneNames } from '@rf2/shared/utility';
import { render } from '@testing-library/react';
import React from 'react';
import { ItemQuantities } from './ItemQuantities';

const toteItem = {
  id: '28391',
  toteId: '1',
  itemIndex: 0,
  orderNo: '1',
  articleId: '28391',
  aisle: '9',
  bay: '56',
  shelf: '2',
  orderedQuantity: 3,
  currentPage: 1,
  totalPages: 3,
  stockOnHand: 3,
  article: {
    id: '28391',
    articleId: '28391',
    brand: 'Woolworths',
    description: 'Woolworths Pumpkin Cubes 480g',
    imageUrl: 'https://uat-assets.woolworths.com.au/images/1005/28391.jpg?impolicy=wowtofzcfpg&w=250&h=250',
    unitPrice: null,
    pricingUnit: null,
    volumeSize: '480',
  },
  totalSuppliedQuantity: 5,
  suppliedDetails: [
    {
      type: 'primary',
      articleId: '28391',
      scanDetails: [
        {
          barcode: '',
          suppliedPrice: 0,
          quantity: 5,
          weight: null,
          batchNo: '',
          expiryDate: '',
          dataBar: '',
        },
      ],
    },
  ],
  secondaryArticleId: '124577',
  version: '0.0.1',
  price: 4.0,
  personalShopperNotes: 'Please only make a substitution if you really have too. As this is my favourite product.',
};

const tote = {
  id: '1',
  orderNo: '1',
  position: 1,
  pickingZoneId: 1,
  pickingZoneName: PickingZoneNames.AMBIENT,
  barcode: '140031688003',
  version: '1',
};

const order = {
  id: '1',
  packagingType: PackagingTypes.PLASTIC,
  deliveryMethod: 'COLLECT',
  locationCode: 5,
  deliveryDate: '2021-03-18T16:00:00',
  customerName: 'Dave',
};

describe('render ItemQuantities', () => {
  it('should render with an each pricing unit', async () => {
    const { container } = render(<ItemQuantities toteItem={toteItem} order={order} tote={tote} showExpiry={true} />);

    expect(container).toMatchSnapshot();
  });

  it('should render with a KG pricing unit', async () => {
    const { container } = render(
      <ItemQuantities
        toteItem={{ ...toteItem, article: { ...toteItem.article, pricingUnit: 'KG' } }}
        order={order}
        tote={tote}
        showExpiry={true}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should display tote position and plastic bag', async () => {
    const { findByText } = render(
      <ItemQuantities
        toteItem={{ ...toteItem }}
        order={{ ...order, packagingType: 'ONLINE BAG' }}
        tote={tote}
        showExpiry={true}
      />
    );

    const toteNode = await findByText(/tote/i);
    const toteValue = toteNode.nextSibling;
    expect(toteValue).toHaveTextContent(/1/i);
    expect(toteValue).toHaveTextContent(/plastic/i);
  });

  it('should display tote position and paper bag', async () => {
    const { findByText } = render(
      <ItemQuantities
        toteItem={{ ...toteItem }}
        order={{ ...order, packagingType: PackagingTypes.PAPER }}
        tote={{ ...tote, position: 2 }}
        showExpiry={true}
      />
    );

    const toteNode = await findByText(/tote/i);
    const toteValue = toteNode.nextSibling;
    expect(toteValue).toHaveTextContent(/2/i);
    expect(toteValue).toHaveTextContent(/paper/i);
  });
});
