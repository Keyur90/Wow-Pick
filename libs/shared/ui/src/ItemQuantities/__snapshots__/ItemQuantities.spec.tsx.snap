// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ItemQuantities should render with a KG pricing unit 1`] = `
<div>
  <div
    class="makeStyles-root-14 makeStyles-root-16"
  >
    <div
      class="makeStyles-root-17"
    >
      <div
        class="makeStyles-label-18 makeStyles-label-20"
      >
        ORD (KG)
      </div>
      <div
        class="makeStyles-value-19"
      >
        3
      </div>
    </div>
    <div
      class="makeStyles-root-17"
    >
      <div
        class="makeStyles-label-18 makeStyles-label-21"
      >
        SUP
      </div>
      <div
        class="makeStyles-value-19"
      >
        3
      </div>
    </div>
    <div
      class="makeStyles-root-17"
    >
      <div
        class="makeStyles-label-18 makeStyles-label-22"
      >
        SOH
      </div>
      <div
        class="makeStyles-value-19"
      >
        3
      </div>
    </div>
    <div
      class="makeStyles-root-17"
    >
      <div
        class="makeStyles-label-18 makeStyles-label-23"
      >
        TOTE
      </div>
      <div
        class="makeStyles-value-19"
      >
        <div
          class="makeStyles-toteStatus-15"
        >
          1
          <span
            class="makeStyles-root-25 makeStyles-root-26"
          >
            <span
              class="MuiTypography-root makeStyles-statusText-24 MuiTypography-body1 MuiTypography-noWrap"
            >
              Plastic
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`render ItemQuantities should render with an each pricing unit 1`] = `
<div>
  <div
    class="makeStyles-root-1 makeStyles-root-3"
  >
    <div
      class="makeStyles-root-4"
    >
      <div
        class="makeStyles-label-5 makeStyles-label-7"
      >
        ORD (KG)
      </div>
      <div
        class="makeStyles-value-6"
      >
        3
      </div>
    </div>
    <div
      class="makeStyles-root-4"
    >
      <div
        class="makeStyles-label-5 makeStyles-label-8"
      >
        SUP
      </div>
      <div
        class="makeStyles-value-6"
      >
        3
      </div>
    </div>
    <div
      class="makeStyles-root-4"
    >
      <div
        class="makeStyles-label-5 makeStyles-label-9"
      >
        SOH
      </div>
      <div
        class="makeStyles-value-6"
      >
        3
      </div>
    </div>
    <div
      class="makeStyles-root-4"
    >
      <div
        class="makeStyles-label-5 makeStyles-label-10"
      >
        TOTE
      </div>
      <div
        class="makeStyles-value-6"
      >
        <div
          class="makeStyles-toteStatus-2"
        >
          1
          <span
            class="makeStyles-root-12 makeStyles-root-13"
          >
            <span
              class="MuiTypography-root makeStyles-statusText-11 MuiTypography-body1 MuiTypography-noWrap"
            >
              Plastic
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
